src/client/grid/hex_name.tsx

```diff
@@ -7,37 +7,24 @@ import { Point, distanceToSide, movePointInRadDirection, pointBetween, polygon }
 interface HexNameProps {
   name: string; center: Point; size: number;
   rotation?: Rotation;
-  isCity?: boolean;
 }
 
-export function HexName(props: HexNameProps) {
-  if (props.rotation != null && props.isCity) return <HexNamePointyTop {...props} />;
-  return <HexNameFlatTop {...props} />;
-}
-
-export function HexNamePointyTop({ name, center, rotation, size }: HexNameProps) {
-  const containerCorners = useMemo(() => {
-    const height = 12;
-    const topLeft = movePointInRadDirection(movePointInRadDirection(center, distanceToSide(size), Math.PI), height, Math.PI / 2);
-    const bottomLeft = movePointInRadDirection(topLeft, height * 2, -Math.PI / 2);
-    const topRight = movePointInRadDirection(movePointInRadDirection(center, distanceToSide(size), 0), height, Math.PI / 2);
-    const bottomRight = movePointInRadDirection(topRight, height * 2, -Math.PI / 2);
+let nextHexNamePathId = 0;
 
-    return polygon([
-      topLeft,
-      bottomLeft,
-      bottomRight,
-      topRight,
-    ]);
-  }, [center, size]);
-  return <>
-    <Rotate rotation={rotation} center={center} reverse={true}>
-      <polygon points={containerCorners} className={styles.hexNameContainer} strokeWidth="1" />
-      <text x={center.x} y={center.y} dominantBaseline="middle" textAnchor="middle">{name}</text>
-    </Rotate>
-  </>;
+export function HexName(props: HexNameProps) {
+  let pathId = nextHexNamePathId;
+  nextHexNamePathId += 1;
```

Comment:
How does this work? I don't understand, every render, we're assigning a new pathId to the given element?

If we package this in a useMemo, would that work?

const pathId = useMemo(() => ++nextHexNamePathId, []);

That way it never gets updated once set for a given hex.

Reply:
Or maybe useId?

https://react.dev/reference/react/useId

Reply:
Nice, `useId` is a perfect solution for this. (The requirement just being that there's a unique value that can be passed into the two child attributes.)